astropy/io/ascii/tests/test_write.py

```diff
@@ -629,3 +639,41 @@ def test_write_overwrite_ascii(format, fast_writer, tmpdir):
                 fast_writer=fast_writer)
         t.write(fp, overwrite=True, format=format,
                 fast_writer=fast_writer)
+
+
+fmt_name_classes = list(chain(ascii.core.FAST_CLASSES.items(),
+                              ascii.core.FORMAT_CLASSES.items()))
+@pytest.mark.parametrize("fmt_name_class", fmt_name_classes)
+def test_roundtrip_masked(fmt_name_class):
+    """
+    Round trip a simple masked table through every writable format and confirm
+    that reading back gives the same result.
+    """
+    fmt_name, fmt_cls = fmt_name_class
+
+    if not getattr(fmt_cls, '_io_registry_can_write', True):
+        return
+
+    # Skip tests for fixed_width or HTML without bs4
+    if ((fmt_name == 'html' and not HAS_BEAUTIFUL_SOUP)
```

Comment:
Cool.  In the interest of not further clogging CI, I would like to defer this change for now.  Anyway it should probably be done as a global change across astropy if we decide this is now the preferred idiom.  See #5543.